docs/models_dev_guide.md

```diff
@@ -0,0 +1,263 @@
+
+```py
+etam.list_models()
+# ["model@0.1", "model@1.0", "model@2.0"]
+
+etam.download_model("model@0.1")
+# model@0.1 is downloaded
+
+etam.download_model("model")
+# model@2.0 is downloaded (the latest version)
+```
+
+#### Locating and loading models
+
+You can obtain the local path to a model by running:
+
+```py
+model_path = etam.find_model(name)
+```
+
+Note that the model will not yet exist at the returned model path if you
+haven't downloaded it.
+
+Although you are free to manually load a model from its `model_path`, the ETA
+library provides several subclasses of `eta.core.models.ModelWeights` that can
+automatically load a model given only its `name`. See the
+[Model weights](#model-weights) section for more information.
+
+#### Publishing a new model
+
+To publish a new model, you can follow the general receipt:
+
+```py
+#
+# Full workflow for publishing a new model
+#
+
+# Recommend paths for the given model by looking for older versions
+# of the model in the model manifests
+base_filename, models_dir = etam.recommend_paths_for_model(name)
+
+# Perform a dry run of the model registration to check for any errors
+# before uploading the model to the cloud
+etam.register_model_dry_run(name, base_filename, models_dir)
+
+# Upload the file to remote storage
+# Then instantiate the relevant ModelManager describing its location
+manager = ...
+
+# Register the model
+etam.register_model(name, base_filename, models_dir, manager)
+```
+
+The above process can be completely automated for custom cloud storage
+applications, including the uploading of the model to cloud storage and the
+generation of the relevant model manager instance describing the model.
+To do this, one should implement a new subclass of
+`eta.core.models.ModelManager`.
+
+> Note: Only Voxel51 administrators can upload models to cloud storage, so
+> one must submit a request to have the ETAModelManager instance generated for
+> your model
+
+#### Flushing local models
+
+Over time you may accumulate too many local copies of models. To clear your
+local disk space, you can flush local models in various ways:
+
+```py
+# Flush a specific model from local storage
+etam.flush_model(model)
+
+# Flush an entire models directory
+etam.flush_models_directory(models_dir)
+
+# Flush all local models
+etam.flush_all_models()
+```
+
+Remember that flushing is a reversible operation; you can always re-download
+all models from cloud storage at any time.
+
+#### Managing the lifecycle of versioned models
+
+The ETA config provides a `max_model_versions_to_keep` attribute that allows
+you to configure the maximum number of versions of a given model to keep on
+disk. If this limit is execeeded, the oldest versions of a model are deleted
+until the limit is satisifed again.
+
+The models framework automatically checks this condition and performs the
+necessary cleanup after each time a model is downloaded. However, you can
+manually invoke the cleanup at any time by running:
+
+```py
```

Comment:
Maybe this should be performed during initialization or cleanup of ETA?  It seems risky to require an interactive function call to manage model lifecycle.  (I'm envisioning long-running installations...)

Reply:
Yeah good point- on startup is definitely preferable to mid-download. I'll update

Reply:
Done